### PR TITLE
zmqpp: support conan v2 more, add patch description

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zmqpp/***

Try to close https://github.com/conan-io/conan-center-index/issues/19607.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
Looks good, thanks :)

File: recipes/zmqpp/all/conanfile.py [hunk 1]
```diff
@@ -37,24 +41,20 @@ def config_options(self):
...
-            except Exception:
-                pass
+            self.options.rm_safe("fPIC")
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("zeromq/4.3.4")
+        self.requires("zeromq/4.3.4", transitive_headers=True)
```
Comment (SpaceIm) [inline recipes/zmqpp/all/conanfile.py:50]:
```suggestion
        self.requires("zeromq/4.3.4", transitive_headers=True, transitive_libs=True)
```

https://github.com/zeromq/zmqpp/blob/4.2.0/src/zmqpp/context.hpp#L76-L80 for example

Comment (toge) [reply recipes/zmqpp/all/conanfile.py]: @SpaceIm 
Thank you for your detail information.
Now I see.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**